platform/posix/ota_pal/source/ota_pal_posix.c

```diff
@@ -0,0 +1,626 @@
+    }
+    else
+    {
+        LogError( ( "Failed to read signer cert." ) );
+    }
+
+    BIO_free_all( pBio );
+    X509_free( pCert );
+
+    /* pPkey should be freed by the caller */
+    return pPkey;
+}
+/*-----------------------------------------------------------*/
+
+
+/* Attempt to create a new receive file for the file chunks as they come in. */
+OtaPalStatus_t otaPal_CreateFileForRx( OtaFileContext_t * const C )
+{
+    OtaPalStatus_t result = OTA_PAL_COMBINE_ERR( OtaPalUninitialized, 0 );
+    char realFilePath[ OTA_FILE_PATH_LENGTH_MAX ];
+
+    if( C != NULL )
+    {
+        if( C->pFilePath != NULL )
+        {
+            if( C->pFilePath[ 0 ] != '/' )
+            {
+                int res = snprintf( realFilePath, OTA_FILE_PATH_LENGTH_MAX, "%s/%s", getenv( "PWD" ), C->pFilePath );
+                assert( res >= 0 );
+                ( void ) res; /* Suppress the unused variable warning when assert is off. */
+            }
+            else
+            {
+                strncpy( realFilePath, ( const char * ) C->pFilePath, strlen( ( const char * ) C->pFilePath ) + 1 );
+            }
+
+            /* Linux port using standard library */
+            /* coverity[misra_c_2012_rule_21_6_violation] */
+            C->pFile = fopen( ( const char * ) realFilePath, "w+b" ); /*lint !e586
+                                                                       * C standard library call is being used for portability. */
+
+            if( C->pFile != NULL )
+            {
+                result = OTA_PAL_COMBINE_ERR( OtaPalSuccess, 0 );
+                LogInfo( ( "Receive file created." ) );
+            }
+            else
+            {
+                result = OTA_PAL_COMBINE_ERR( OtaPalRxFileCreateFailed, errno );
+                LogError( ( "Failed to start operation: Operation already started. failed to open -- %s Path ", C->pFilePath ) );
+            }
+        }
+        else
+        {
+            result = OTA_PAL_COMBINE_ERR( OtaPalRxFileCreateFailed, 0 );
+            LogError( ( "Invalid file path provided." ) );
+        }
+    }
+    else
+    {
+        result = OTA_PAL_COMBINE_ERR( OtaPalRxFileCreateFailed, 0 );
+        LogError( ( "Invalid context provided." ) );
+    }
+
+    /* Exiting function without calling fclose. Context file handle state is managed by this API. */
+    return result;
+}
+
+
+/* Abort receiving the specified OTA update by closing the file. */
+OtaPalStatus_t otaPal_Abort( OtaFileContext_t * const C )
+{
+    /* Set default return status to uninitialized. */
+    OtaPalMainStatus_t mainErr = OtaPalUninitialized;
+    OtaPalSubStatus_t subErr = 0;
+    int32_t lFileClosresult;
+
+    if( NULL != C )
+    {
+        /* Close the OTA update file if it's open. */
+        if( NULL != C->pFile )
+        {
+            /* Linux port using standard library */
+            /* coverity[misra_c_2012_rule_21_6_violation] */
+            lFileClosresult = fclose( C->pFile ); /*lint !e482 !e586
```

Comment:
We do not need these PCLint comments anymore as we are using coverity now. Applies to other places as well.